app/models/miq_region.rb

```diff
@@ -19,6 +19,10 @@ class MiqRegion < ApplicationRecord
   virtual_has_many :miq_templates
   virtual_has_many :vms
 
+  virtual_aggregate :aggregate_physical_cpus, :host_hardwares, :sum, :cpu_sockets
+  has_many :host_hardwares, :class_name => 'Hardware', :through => :hosts, :source => :hardware
```

Comment:
Can this be defined after it is called by the `virtual_aggregate`?

Should the `has_many` calls maybe be defined before any `virtual_*` stuff is defined?